go/v1beta1/storage/esutil/client.go

```diff
@@ -0,0 +1,532 @@
+	Index      string
+	DocumentId string
+	Refresh    string // TODO: use RefreshOption type
+}
+
+type DeleteRequest struct {
+	Index   string
+	Search  *EsSearch
+	Refresh string // TODO: use RefreshOption type
+}
+
+const defaultPitKeepAlive = "5m"
+const grafeasMaxPageSize = 1000
+
+type Client interface {
+	Create(ctx context.Context, request *CreateRequest) (string, error)
+	BulkCreate(ctx context.Context, request *BulkCreateRequest) (*EsBulkResponse, error)
+	Search(ctx context.Context, request *SearchRequest) (*SearchResponse, error)
+	MultiSearch(ctx context.Context, request *MultiSearchRequest) (*EsMultiSearchResponse, error)
+	MultiGet(ctx context.Context, request *MultiGetRequest) (*EsMultiGetResponse, error)
+	List(ctx context.Context, request *ListRequest) (*ListResponse, error)
+	Update(ctx context.Context, request *UpdateRequest, message proto.Message) error
+	Delete(ctx context.Context, request *DeleteRequest) error
+}
+
+type client struct {
+	logger   *zap.Logger
+	esClient *elasticsearch.Client
+	filterer filtering.Filterer
+}
+
+func NewClient(logger *zap.Logger, esClient *elasticsearch.Client, filterer filtering.Filterer) Client {
+	return &client{
+		logger,
+		esClient,
+		filterer,
+	}
+}
+
+func (c *client) Create(ctx context.Context, request *CreateRequest) (string, error) {
+	log := c.logger.Named("Create")
+	str, err := protojson.MarshalOptions{EmitUnpopulated: true}.Marshal(request.Message)
+	if err != nil {
+		return "", err
+	}
+
+	if request.Refresh == "" {
+		request.Refresh = "true"
+	}
+
+	indexOpts := []func(*esapi.IndexRequest){
+		c.esClient.Index.WithContext(ctx),
+		c.esClient.Index.WithRefresh(request.Refresh),
+	}
+	if request.DocumentId != "" {
+		indexOpts = append(indexOpts, c.esClient.Index.WithDocumentID(request.DocumentId))
+	}
+
+	res, err := c.esClient.Index(
+		request.Index,
+		bytes.NewReader(str),
+		indexOpts...,
+	)
+	if err != nil {
+		return "", err
+	}
+	if res.IsError() {
+		return "", errors.New(fmt.Sprintf("unexpected response from elasticsearch: %s", res.String()))
+	}
+
+	esResponse := EsIndexDocResponse{}
+	if err := DecodeResponse(res.Body, &esResponse); err != nil {
+		return "", err
+	}
+
+	log.Debug("elasticsearch response", zap.Any("response", esResponse))
+
+	return esResponse.Id, nil
+}
+
+func (c *client) BulkCreate(ctx context.Context, request *BulkCreateRequest) (*EsBulkResponse, error) {
+	log := c.logger.Named("BulkCreate")
+
+	// build the request body using newline delimited JSON (ndjson)
+	// each message is represented by two JSON structures:
+	// the first is the metadata that represents the ES operation, in this case "index"
+	// the second is the source payload to index
+	// in total, this body will consist of (len(messages) * 2) JSON structures, separated by newlines, with a trailing newline at the end
+	var body bytes.Buffer
+	for _, item := range request.Items {
+		metadata := &EsBulkQueryFragment{
+			Index: &EsBulkQueryIndexFragment{
```

Comment:
Setting `index` might have to be the `else` case for that conditional, not sure what ES does when there are multiple actions in the same block.

Reply:
I can remove this and instead use `WithIndex` when making the request.  this should be fine since we're already using the same index for each of these documents anyways.